docs/developers/components/managing.md

```diff
@@ -0,0 +1,97 @@
+
+> For a useful shortcut on UNIX machines run: `harperdb get_configuration json=true | jq ".rootPath" | sed 's/"//g'`
+
+This path is the Harper instance. Within this directory, locate the root config titled **harperdb-config.yaml**, and the components root path. The components root path will be `<rootPath>/components` by default, but it can also be configured. If necessary, use `harperdb get_configuration` again and look for the **componentsRoot** field for the exact path.
+
+### Adding components to root
+
+Similar to how components can specify other components within their **config.yaml**, components can be added to Harper by adding them to the **harperdb-config.yaml**.
+
+The configuration is very similar to that of **config.yaml**. Entries are comprised of a top-level `<name>:`, and an indented `package: <specifier>` field. Any additional component option can also be included as indented fields.
+
+```yaml
+status-check:
+  package: "@harperdb/status-check"
+```
+
+The key difference between this and a component's **config.yaml** is that the name does **not** need to be associated with a **package.json** dependency. When Harper starts up, it transforms these configurations into a **package.json** file, and then executes a form of `npm install`. Thus, the `package: <specifier>` can be any valid dependency syntax such as npm packages, GitHub repos, tarballs, and local directories are all supported.
+
+Given a root config like:
+
+```yaml
+myGithubComponent:
+  package: HarperDB-Add-Ons/package#v2.2.0 # install from GitHub 
+myNPMComponent:
+  package: harperdb # install from npm
+myTarBall:
+  package: /Users/harper/cool-component.tar # install from tarball
+myLocal:
+  package: /Users/harper/local # install from local path
+myWebsite:
+  package: https://harperdb-component # install from URL
+```
+
+Harper will generate a **package.json** like:
+
+```json
+{
+  "dependencies": {
+    "myGithubComponent": "github:HarperDB-Add-Ons/package#v2.2.0",
+    "myNPMComponent": "npm:harperdb",
+    "myTarBall": "file:/Users/harper/cool-component.tar",
+    "myLocal": "file:/Users/harper/local",
+    "myWebsite": "https://harperdb-component"
+  }
+}
+```
+
+npm will install all the components and store them in `<rootPath>/<componentsRoot>`. A symlink back to `<rootPath>/node_modules` is also created for dependency resolution purposes.
+
+The package prefix is automatically added, however you can manually set it in your package reference.
+
+```yaml
+myCoolComponent:
+  package: file:/Users/harper/cool-component.tar
+```
+
+By specifying a file path, Harper _should_ generate a symlink from that location to the **componentsRoot**. In case the symlink is not being generated, the final, advanced method for component loading is to manually create symlinks from the local component source directory to the Harper instance **componentsRoot**. For most UNIX machines this is as simple as: `ln -s <path/to/component/source> <componentsRoot>`.
```

Comment:
I'm not sure I follow why the symlink wouldn't be created. Is that a bug?

The second sentence reads weird. I don't think you need the whole "final, advanced method" wording.

```suggestion
By specifying a file path, Harper _should_ generate a symlink from that location to the `componentsRoot`. In case the symlink is not being generated, manually create the symlink from the local component source directory to the Harper instance `componentsRoot*`. For most UNIX machines this is as simple as: `ln -s <path/to/component/source> <componentsRoot>`.
```

Reply:
You are right. I think I'm just going to remove this. We can consider npm not symlinking things as a bug.